cassandane/Cassandane/Test/NewTestUrl.pm

```diff
@@ -0,0 +1,132 @@
+#!/usr/bin/perl
+
+package Cassandane::Test::NewTestUrl;
+use strict;
+use warnings;
+
+use JSON;
+use LWP::UserAgent;
+
+use base qw(Cassandane::Unit::TestCase);
+
+sub new
+{
+    my $class = shift;
+    my $self = $class->SUPER::new(@_);
+    return $self;
+}
+
+sub test_basic
+{
+    my ($self) = @_;
+
+    my $ts = bless {}, 'Cassandane::Unit::TestCase';
+
+    my $string_based_url = $ts->new_test_url("string based");
+    my $code_based_url = $ts->new_test_url(sub {
```

Comment:
Is there a reason you're creating `$ts`, and not just using `$self`, which already is a Cassandane::Unit::TestCase subclass, and so inherits `new_test_url`?

Someday when I end up needing this thing myself, I'm going to come looking at these tests as an example of how to drive it, and this is gonna confuse me...

I guess maybe you're trying to isolate the thing under test from the test that's testing it?  Which I think makes some sense, but if so it's tricky and could use a comment.

In practice, with `sub new_test_url` being defined as a method on Cassandane::Unit::TestCase, any test can simply call it like `$self->new_test_url(...)`.  But someone coming here for an example may just copy what's here, not realising it doesn't have to be that hard.  Or, realise it could be simpler, and get stuck second guessing why it's not. ;)

Reply:
Oops. Hold over from when this was in t/.

Fixed!